src/NuGetForUnity/Editor/PackageSource/NugetApiClientV3.cs

```diff
@@ -414,26 +463,62 @@ private async void InitializeApiAddresses(NugetPackageSourceV3 packageSource)
                     }
                 }
 
-                if (string.IsNullOrEmpty(packageBaseAddress))
+                if (!string.IsNullOrEmpty(PackageDownloadUrlTemplateOverwrite))
+                {
+                    packageDownloadUrlTemplate = PackageDownloadUrlTemplateOverwrite;
+                }
+                else if (string.IsNullOrEmpty(packageBaseAddress))
+                {
+                    UnityMainThreadDispatcher.Dispatch(
+                        () =>
+                        {
+                            var displayDialog = EditorUtility.DisplayDialog(
+                                "Missing 'PackageBaseAddress' endpoint",
+                                $"The used NuGet V3 API '{apiIndexJsonUrl}' doesn't support the 'PackageBaseAddress' endpoint. You need to provide the url manually by specifying '{NugetConfigFile.PackageDownloadUrlTemplateOverwriteAttributeName}' on the package source. If this is a NuGet source provided by Artifactory you can click on 'Configure for Artifactory' to configure it.",
+                                "Configure for Artifactory",
+                                "Cancel");
+                            if (displayDialog)
+                            {
+                                packageDownloadUrlTemplate = $"{registrationsBaseUrl}Download/{{0}}/{{1}}";
+                                PackageDownloadUrlTemplateOverwrite = packageDownloadUrlTemplate;
+                                packageSource.UpdateSearchBatchSize =
+                                    1; // Artifactory somehow can't handle search queries containing multiple packageId's.
```

Comment:
I was hoping you will find some magic incantation to make it work :). I'll see if I can open an issue to Artifactory about this tomorrow.